services/ux-backend/handlers/onboardingtokens/handler.go

```diff
@@ -52,3 +59,35 @@ func handleUnsupportedMethod(w http.ResponseWriter, r *http.Request) {
 		klog.Errorf("failed write data to response writer: %v", err)
 	}
 }
+
+func getQuotaInGib(r *http.Request) (*int, error) {
+	// When ContentLength is 0 that means request body is empty and
+	// storage quota is unlimited
+	if r.ContentLength == 0 {
+		return nil, nil
+	}
+	var quota = struct {
+		Value int    `json:"value"`
+		Unit  string `json:"unit"`
+	}{}
+	err := json.NewDecoder(r.Body).Decode(&quota)
+	if err != nil {
+		return nil, err
+	}
+	if quota.Value <= 0 {
+		return nil, fmt.Errorf("invalid Value sent in request body: %v", quota.Value)
+	}
+	if quota.Unit == "" {
+		return nil, fmt.Errorf("invalid Unit type sent in request body, Valid types are [Gi,Ti,Pi]: %v", quota.Unit)
+	}
+	UnitToGib := map[string]int{
+		"Gi": 1,
+		"Ti": 1024,
+		"Pi": 1048576,
```

Comment:
Please communicate how you got to the number  1048576. There are 2 options:
```suggestion
		"Pi": 1024 * 1024,
```

or 

```suggestion
		"Pi": 1048576, // 1024 * 1024
```